core/database.py

```diff
@@ -83,20 +84,15 @@ async def _token_task(self) -> None:
 
             await asyncio.sleep(self._gist_timeout)
 
-    def _handle_discord_tokens(self, *bodies: dict[str, str], paste_id: str) -> None:
-        formatted_bodies = "\n".join(b["content"] for b in bodies)
-
-        tokens = list(DISCORD_TOKEN_REGEX.finditer(formatted_bodies))
-
+    def _handle_discord_tokens(self, tokens: list[str], paste_id: str) -> None:
         if not tokens:
```

Comment:
Not entirely related but tokens should only be stored for upload if the feature is active:
```suggestion
        if not self._handling_tokens or not tokens:
```